pkg/apis/controlplane/v1beta2/types.go

```diff
@@ -0,0 +1,292 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	secv1alpha1 "github.com/vmware-tanzu/antrea/pkg/apis/security/v1alpha1"
+	statsv1alpha1 "github.com/vmware-tanzu/antrea/pkg/apis/stats/v1alpha1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +genclient:onlyVerbs=list,get,watch
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// AppliedToGroup is the message format of antrea/pkg/controller/types.AppliedToGroup in an API response.
+type AppliedToGroup struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// Pods is a list of Pods selected by this group.
+	Pods []GroupMemberPod `json:"pods,omitempty" protobuf:"bytes,2,rep,name=pods"`
+	// GroupMembers is list of resources selected by this group. This eventually will replace Pods
+	GroupMembers []GroupMember `json:"groupMembers,omitempty" protobuf:"bytes,3,rep,name=groupMembers"`
+}
+
+// PodReference represents a Pod Reference.
+type PodReference struct {
+	// The name of this pod.
+	Name string `json:"name,omitempty" protobuf:"bytes,1,opt,name=name"`
+	// The namespace of this pod.
+	Namespace string `json:"namespace,omitempty" protobuf:"bytes,2,opt,name=namespace"`
+}
+
+// NamedPort represents a Port with a name on Pod.
+type NamedPort struct {
+	// Port represents the Port number.
+	Port int32 `json:"port,omitempty" protobuf:"varint,1,opt,name=port"`
+	// Name represents the associated name with this Port number.
+	Name string `json:"name,omitempty" protobuf:"bytes,2,opt,name=name"`
+	// Protocol for port. Must be UDP, TCP, or SCTP.
+	Protocol Protocol `json:"protocol,omitempty" protobuf:"bytes,3,opt,name=protocol"`
+}
+
+// GroupMemberPod represents a GroupMember related to Pods.
+type GroupMemberPod struct {
```

Comment:
Do you plan to remove this in this PR or another one? BTW, I think we could just remove `GroupMemberPod` in AppliedToGroup as well, otherwise it doesn't make sense to add `GroupMember` to it in the beginning. 
@jianjuns @antoninbas @abhiraut

Reply:
Yes I will push a separate patch based on this PR shortly... removing GroupMemberPod is a big change itself as it affects quite a lot files in both controller and agent.